charmdir.go

```diff
@@ -172,10 +173,15 @@ func resolveSymlinkedRoot(rootPath string) (string, error) {
 // ArchiveTo creates a charm file from the charm expanded in dir.
 // By convention a charm archive should have a ".charm" suffix.
 func (dir *CharmDir) ArchiveTo(w io.Writer) error {
-	return writeArchive(w, dir.Path, dir.revision, dir.Meta().Hooks())
+	versionString, err := MaybeCreateGenerateString(dir.Path)
+	if err != nil {
+		logger.Debugf("version file creation/overwrite failed : %v", err)
```

Comment:
debug message is incorrect

Reply:
if there really is an error, we should log at Error level

Reply:
Done